test/unit_tests/threads/test_multicore.cpp

```diff
@@ -0,0 +1,248 @@
+            for (auto i = 0; i < cache_sim_times.size(); ++i) {
+                std::cout << nof_threads_range[i] << "\t" << no_cache_sim_times[i] << "\t"
+                          << cache_sim_times[i] << std::endl;
+            }
+        }
+        THEN("we check that the more threads we have the faster the simulation runs") {
+            if (nof_threads_range.size() > 2) {
+                REQUIRE(std::is_sorted(cache_sim_times.rbegin(), cache_sim_times.rend()));
+                REQUIRE(std::is_sorted(no_cache_sim_times.rbegin(), no_cache_sim_times.rend()));
+                REQUIRE(cache_sim_times < no_cache_sim_times);
+                THEN(
+                    "we check that the standard deviaton is above 25% from the mean for simulation "
+                    "vectors") {
+                    const auto cache_mean =
+                        std::accumulate(cache_sim_times.begin(), cache_sim_times.end(), 0.0) /
+                        cache_sim_times.size();
+                    const auto cache_std_dev = std::sqrt(
+                        std::accumulate(cache_sim_times.begin(),
+                                        cache_sim_times.end(),
+                                        0.0,
+                                        [cache_mean](double a, double b) {
+                                            return a + (b - cache_mean) * (b - cache_mean);
+                                        }) /
+                        cache_sim_times.size());
+                    const auto no_cache_mean =
+                        std::accumulate(no_cache_sim_times.begin(), no_cache_sim_times.end(), 0.0) /
+                        no_cache_sim_times.size();
+                    const auto no_cache_std_dev = std::sqrt(
+                        std::accumulate(no_cache_sim_times.begin(),
+                                        no_cache_sim_times.end(),
+                                        0.0,
+                                        [no_cache_mean](double a, double b) {
+                                            return a + (b - no_cache_mean) * (b - no_cache_mean);
+                                        }) /
+                        no_cache_sim_times.size());
+                    REQUIRE(cache_std_dev / cache_mean > 0.25);
+                    REQUIRE(no_cache_std_dev / no_cache_mean > 0.25);
+                    // print the standard deviations
+                    std::cout << "[parallel][cache][standard deviation] : " << cache_std_dev
+                              << std::endl;
+                    std::cout << "[parallel][no-cache][standard deviation] : " << no_cache_std_dev
+                              << std::endl;
+                }
+            } else {
+                WARN("Not enough threads to test parallel performance KPI");
+            }
+        }
+    }
+
+    SECTION("Test serial mode", "[NEURON][multicore][serial]") {
+        GIVEN("cache efficient is set to 1") {
```

Comment:
Any particular reason not to have `auto cache_efficient = GENERATE(0, 1);` here too?